Fakebook.Posts/Fakebook.Posts.IntegrationTests/ClientActions/GetPostTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Fakebook.Posts.Domain.Interfaces;
+using Fakebook.Posts.Domain.Models;
+using Fakebook.Posts.IntegrationTests.Services;
+using Fakebook.Posts.RestApi;
+using Fakebook.Posts.RestApi.Services;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Fakebook.Posts.IntegrationTests.ClientActions
+{
+    public class GetPostTests : IClassFixture<WebApplicationFactory<Startup>>
+    {
+        private readonly WebApplicationFactory<Startup> _factory;
+
+        public GetPostTests(WebApplicationFactory<Startup> factory)
+        {
+            _factory = factory;
+        }
+
+        /// <summary>
+        /// Tests the PostsController class' GetAsync method. Ensures that a proper post id results in status 200 Ok.
+        /// </summary>
+        [Fact]
+        public async Task GetAsync_ValidId_ReturnsPost()
+        {
+            // Arrange
+            Mock<IPostsRepository> mockRepo = new();
+            Mock<IFollowsRepository> mockFollowRepo = new();
+            Mock<IBlobService> mockBlobService = new();
+            Post post = new Post("test.user@email.com", "test content") { Id = 1 };
+            mockRepo.Setup(r => r.GetAsync(It.IsAny<int>()))
+                .ReturnsAsync(post);
+
+            var client = _factory.WithWebHostBuilder(builder =>
+            {
+                builder.ConfigureTestServices(services =>
+                {
+                    services.AddScoped(sp => mockRepo.Object);
+                    services.AddScoped(sp => mockFollowRepo.Object);
+                    services.AddTransient(sp => mockBlobService.Object);
+                    services.AddAuthentication("Test")
+                        .AddScheme<AuthenticationSchemeOptions, TestAuthHandler>("Test", options => { });
+                });
+            }).CreateClient();
+
+            client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Test");
```

Comment:
just use a using directive here, and don't have so much code duplication
if existing tests have the same duplication, well, that should have already been addressed by now since it's a very simple refactor. up to you if it's part of this PR or not.

Reply:
I will look into fixing it later as this PR is somewhat urgent due to the fix being needed by someone else.